Libs/DICOM/Core/ctkDICOMDisplayedFieldGeneratorLastStudyDateRule.cpp

```diff
@@ -0,0 +1,141 @@
+/*=========================================================================
+
+  Library:   CTK
+
+  Copyright (c) Pixel Medical 2020
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0.txt
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+=========================================================================*/
+
+#include "ctkDICOMDisplayedFieldGeneratorLastStudyDateRule.h"
+
+// ctkDICOM includes
+#include "ctkDICOMDatabase.h"
+#include "ctkDICOMItem.h"
+
+#include <QDebug> //TODO: Remove
+#include <QSqlError>
+#include <QSqlQuery>
+
+//------------------------------------------------------------------------------
+ctkDICOMDisplayedFieldGeneratorLastStudyDateRule::ctkDICOMDisplayedFieldGeneratorLastStudyDateRule()
+{
+}
+
+//------------------------------------------------------------------------------
+QStringList ctkDICOMDisplayedFieldGeneratorLastStudyDateRule::getRequiredDICOMTags()
+{
+  QStringList requiredTags;
+
+  requiredTags << dicomTagToString(DCM_SOPInstanceUID);
+
+  requiredTags << dicomTagToString(DCM_PatientID);
+  requiredTags << dicomTagToString(DCM_PatientName);
+  requiredTags << dicomTagToString(DCM_PatientBirthDate);
+
+  requiredTags << dicomTagToString(DCM_StudyDate);
+
+  return requiredTags;
+}
+
+//------------------------------------------------------------------------------
+void ctkDICOMDisplayedFieldGeneratorLastStudyDateRule::registerEmptyFieldNames(
+  QMap<QString, QString> emptyFieldsDisplaySeries,
+  QMap<QString, QString> emptyFieldsDisplayStudies,
+  QMap<QString, QString> emptyFieldsDisplayPatients )
+{
+  Q_UNUSED(emptyFieldsDisplaySeries);
+  Q_UNUSED(emptyFieldsDisplayStudies);
+  Q_UNUSED(emptyFieldsDisplayPatients);
+}
+
+
+//------------------------------------------------------------------------------
+QString ctkDICOMDisplayedFieldGeneratorLastStudyDateRule::compositePatientID(const QMap<QString, QString> &cachedTagsForInstance)
```

Comment:
It would be better to call the compositePatientID generation method of the database (to reduce code duplication and ensure consistency in what we consider the same patient).